assets/chat.js

```diff
@@ -1,30 +1,18 @@
 import Chat from "./chat/js/chat";
 import emotes from "./emotes.json";
 
-$.when(
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/me`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/history`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/viewer-states`)
-            .done(res)
-            .fail(() => res([]))
-    )
-).then(
-    (userAndSettings, history, viewerStates) =>
-        (window.__chat__ = new Chat()
-            .withUserAndSettings(userAndSettings)
-            .withEmotes(emotes)
-            .withGui()
-            .withViewerStates(viewerStates)
-            .withHistory(history)
-            .withWhispers()
-            .connect(WEBSOCKET_URI))
-);
+const init = async v => {
+    const userInfo = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/me`).done(res).fail(() => res(null)));
+    const history = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/history`).done(res).fail(() => res(null)));
+    const viewerStates = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/viewer-states`).done(res).fail(() => res(null)));
```

Comment:
the proposed changes just wrap the promises in more promises

the modern js would be something like 
```js
const [me, history, viewerStates] = await Promise.all([
  fetch(`${API_URI}/api/chat/me`).then(res => res.json()),
  fetch(`${API_URI}/api/chat/history`).then(res => res.json()),
  fetch(`${API_URI}/api/chat/viewer-states`).then(res => res.json()),
])
```
but neither of these improve the way the requests run... `$.getJSON` is already async and `$.when` is equivalent to `Promise.all`.